internal/elasticsearch/elasticsearch.go

```
@@ -25,125 +27,210 @@ import (
 		return errgo.Mask(err)
 	}
-	if status != http.StatusOK {
-		return errgo.Newf("ElasticSearch GET response status: %d, body: %s", status, body)
+	return nil
+}
+
+// ListAllIndexes does a GET request to the Database and parses the string
+// response returning a slice containing the name of each index.
+// http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/_list_all_indexes.html
+func (db *Database) ListAllIndexes() ([]string, error) {
+	var result map[string]interface{}
+	if err := db.get(db.url("_aliases"), nil, &result); err != nil {
+		return nil, errgo.Mask(err)
 	}
-	err = json.Unmarshal(body, doc)
-	if err != nil {
-		return errgo.Notef(err, "cannot unmarshal body: %s", body)
+	var indices []string
+	for key := range result {
+		// Some ElasticSearch plugins create indices (e.g. ".marvel...") for their
+		// use.  Ignore any that start with a dot.
+		if !strings.HasPrefix(key, ".") {
+			indices = append(indices, key)
+		}
 	}
-	return nil
+	return indices, nil
 }
 
 // PostDocument creates a new auto id document with the given index and _type
 // and returns the generated id of the document.
 func (db *Database) PostDocument(index, type_ string, doc interface{}) (string, error) {
-	data, err := json.Marshal(doc)
-	if err != nil {
-		return "", errgo.Mask(err)
-	}
-	// CONSIDER: err := db.post(db.url(index, type_), doc, &respData)
-	status, _, body, err := db.request("POST", db.url(index, type_), "application/json", string(data))
-	if err != nil {
+	var resp map[string]interface{}
+	if err := db.post(db.url(index, type_), doc, &resp); err != nil {
 		return "", errgo.Mask(err)
 	}
-	if status != http.StatusCreated {
-		return "", errgo.Newf("ElasticSearch POST response status: %d, body: %s", status, body)
-	}
-	var respdata map[string]interface{}
-	if err := json.Unmarshal(body, &respdata); err != nil {
-		return "", errgo.Notef(err, "cannot unmarshal body")
-	}
-	return respdata["_id"].(string), nil
+	return resp["_id"].(string), nil
```

This will panic if the response isn't the expected string.
Better to return an error in this case, I think.